### PR TITLE
fix: forbid DFLYCLUSTER commads set for emulated cluster mode

### PR DESCRIPTION
**Pull request by BorysTheDev**

### CODE REVIEW

**Review by adiholden — DISMISSED** [review]

File: src/server/cluster/cluster_family.cc [hunk 1]
```diff
@@ -401,8 +403,8 @@ void ClusterFamily::ReadWrite(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void ClusterFamily::DflyCluster(CmdArgList args, ConnectionContext* cntx) {
-  if (!IsClusterEnabledOrEmulated()) {
-    return cntx->SendError(kClusterDisabled);
+  if (!IsClusterEnabled()) {
+    return cntx->SendError("Cluster is disabled. Enabled via passing --cluster_mode=yes");
```
Comment (chakaz) [inline src/server/cluster/cluster_family.cc:407]:
```suggestion
    return cntx->SendError("Cluster is disabled. Use --cluster_mode=yes to enable.");
```

File: src/server/cluster/cluster_family.cc [hunk 2]
```diff
@@ -425,12 +425,8 @@ void ClusterFamily::DflyCluster(CmdArgList args, ConnectionContext* cntx) {
   return cntx->SendError(UnknownSubCmd(sub_cmd, "DFLYCLUSTER"), kSyntaxErrType);
 }
 
-void ClusterFamily::DflyClusterMyId(CmdArgList args, ConnectionContext* cntx) {
-  if (!args.empty()) {
-    return cntx->SendError(WrongNumArgsError("DFLYCLUSTER MYID"));
-  }
```
Comment (chakaz) [inline src/server/cluster/cluster_family.cc:431]: Why did you remove the arguments check?

Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:431]: To be consistent with other CLUSTER commands

Comment (chakaz) [reply src/server/cluster/cluster_family.cc:431]: But this command doesn't accept parameters..
Please do apply this check :)

Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:431]: I will add it to other cluster commands in this case

**Review by chakaz — APPROVED** [review]

**Review by adiholden — CHANGES_REQUESTED** [review]
as discussed find why test_replicate_disconnect_cluster is failing and fix it before you merge the PR

File: src/server/cluster/cluster_family.cc [hunk 3]
```diff
@@ -401,11 +409,15 @@ void ClusterFamily::ReadWrite(CmdArgList args, ConnectionContext* cntx) {
...
+  if (!(IsClusterEnabled() || (IsClusterEmulated() && cntx->journal_emulated))) {
+    return cntx->SendError("Cluster is disabled. Use --cluster_mode=yes to enable.");
   }
 
-  VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  if (cntx->conn()) {
+    VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  } else {
+    VLOG(2) << "Got DFLYCLUSTER command (NO_CLIENT_ID): " << args;
+  }
```
Comment (chakaz) [inline src/server/cluster/cluster_family.cc:420]:
```suggestion
  auto print_conn = [](auto* conn) { return conn ? conn->GetClientId() : "NO_CLIENT_ID" };
  VLOG(2) << "Got DFLYCLUSTER command (" << print_conn(cntx->conn()) << "): " << args;
```

File: src/server/cluster/cluster_family.cc [hunk 4]
```diff
@@ -401,11 +409,15 @@ void ClusterFamily::ReadWrite(CmdArgList args, ConnectionContext* cntx) {
...
-    return cntx->SendError(kClusterDisabled);
+  if (!(IsClusterEnabled() || (IsClusterEmulated() && cntx->journal_emulated))) {
+    return cntx->SendError("Cluster is disabled. Use --cluster_mode=yes to enable.");
   }
 
-  VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  if (cntx->conn()) {
+    VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  } else {
+    VLOG(2) << "Got DFLYCLUSTER command (NO_CLIENT_ID): " << args;
```
Comment (adiholden) [inline src/server/cluster/cluster_family.cc:419]: In which flow do you get null connection?

Comment (adiholden) [reply src/server/cluster/cluster_family.cc:419]: oh this is the journal emulated mode

Comment (adiholden) [reply src/server/cluster/cluster_family.cc:419]: nit: 
client_info = cntx->conn()? ntx->conn()->GetClientId() : "stub";
VLOG(2) << "Got DFLYCLUSTER command (" << client_info << args;

**Review by adiholden — APPROVED** [review]

File: src/server/cluster/cluster_family.cc (hunk 4 above)
Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:419]: in journal_emulated

File: src/server/cluster/cluster_family.cc (hunk 3 above)
Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:420]: it's different types. int and char[]

Comment (chakaz) [reply src/server/cluster/cluster_family.cc:420]: You could return a string.. no need to duplicate code
(a string < 16 bytes will not do dynamic allocations)

Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:420]: Will change it next time when do some changes in this file
